sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineCustomImageImpl.java

```diff
@@ -34,7 +34,9 @@ class VirtualMachineCustomImageImpl
     VirtualMachineCustomImageImpl(final String name, ImageInner innerModel, final ComputeManager computeManager) {
         super(name, innerModel, computeManager);
         // set the default value for the hyper-v generation
-        this.innerModel().withHyperVGeneration(HyperVGenerationTypes.V1);
+        if (this.innerModel().hyperVGeneration() == null) {
+            this.innerModel().withHyperVGeneration(HyperVGenerationTypes.V1);
+        }
```

Comment:
Would it better if we do it here?
https://github.com/Azure/azure-sdk-for-java/blob/eee6fb178fd9aa1f287870cc5e8005d81e7e5462/sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineCustomImagesImpl.java#L37-L39

Reply:
Yeah, sure.